File: src/infra/logger.ts
import chalk from 'chalk';

export enum LogLevel {
    DEBUG = 0,
    INFO = 1,
    SUCCESS = 2,
    WARN = 3,
    ERROR = 4
}

export class Logger {
    private static minLevel: LogLevel = LogLevel.DEBUG;

    private static getTime(): string {
        const now = new Date();
        return now.toISOString().replace('T', ' ').substring(0, 19);
    }

    private static formatMessage(level: string, color: any, scope: string, message: string): string {
        const timeStr = chalk.gray(`[${this.getTime()}]`);
        const levelStr = color(`[${level.padEnd(7)}]`);
        const scopeStr = chalk.cyan(`[${scope.toUpperCase()}]`);
        return `${timeStr} ${levelStr} ${scopeStr} ${message}`;
    }

    static debug(scope: string, message: string) {
        if (this.minLevel <= LogLevel.DEBUG) {
            console.log(this.formatMessage('DEBUG', chalk.magenta, scope, message));
        }
    }

    static info(scope: string, message: string) {
        if (this.minLevel <= LogLevel.INFO) {
            console.log(this.formatMessage('INFO', chalk.blue, scope, message));
        }
    }

    static success(scope: string, message: string) {
        if (this.minLevel <= LogLevel.SUCCESS) {
            console.log(this.formatMessage('SUCCESS', chalk.green, scope, message));
        }
    }

    static warn(scope: string, message: string) {
        if (this.minLevel <= LogLevel.WARN) {
            console.log(this.formatMessage('WARN', chalk.yellow, scope, message));
        }
    }

    static error(scope: string, message: string, error?: any) {
        if (this.minLevel <= LogLevel.ERROR) {
            console.error(this.formatMessage('ERROR', chalk.red, scope, message));
            if (error) {
                if (error instanceof Error) {
                    console.error(chalk.red(error.stack));
                } else {
                    console.error(error);
                }
            }
        }
    }
}

// FIX: createLogger factory — used by audit.ts, tts.ts, monitor.ts, evolution files
export function createLogger(scope: string) {
    return {
        debug:   (msg: string)              => Logger.debug(scope, msg),
        info:    (msg: string)              => Logger.info(scope, msg),
        success: (msg: string)              => Logger.success(scope, msg),
        warn:    (msg: string)              => Logger.warn(scope, msg),
        error:   (msg: string, err?: any)   => Logger.error(scope, msg, err),
    };
}

// FIX: default export — used by server.ts and reminders.ts
export default Logger;

File: pc_agent/utils/logger.py
"""OpenClaw-style colored logger for NEXUM PC Agent."""

import os
import sys
from datetime import datetime

DEBUG = os.environ.get("NEXUM_DEBUG") == "1"

# ANSI colors
RESET  = "\033[0m"
GREY   = "\033[90m"
CYAN   = "\033[96m"
GREEN  = "\033[92m"
YELLOW = "\033[93m"
RED    = "\033[91m"
BOLD   = "\033[1m"

_use_colors = sys.stdout.isatty() or os.environ.get("NEXUM_COLOR") == "1"


def _ts() -> str:
    return datetime.now().strftime("%H:%M:%S.%f")[:-3]


def _fmt(level: str, color: str, msg: str) -> str:
    if _use_colors:
        return f"{GREY}[{_ts()}]{RESET} {color}{BOLD}[{level}]{RESET} {msg}"
    return f"[{_ts()}] [{level}] {msg}"


class Logger:
    def debug(self, msg: str):
        if DEBUG:
            print(_fmt("DEBUG", GREY, msg))

    def info(self, msg: str):
        print(_fmt("INFO", CYAN, msg))

    def success(self, msg: str):
        print(_fmt("OK", GREEN, msg))

    def warn(self, msg: str):
        print(_fmt("WARN", YELLOW, msg), file=sys.stderr)

    def error(self, msg: str):
        print(_fmt("ERROR", RED, msg), file=sys.stderr)


log = Logger()
